src/pip/_internal/pep425tags.py

```diff
@@ -135,7 +135,12 @@ def get_platform():
     if result == "linux_x86_64" and _is_running_32bit():
         # 32 bit Python program (running on a 64 bit Linux): pip should only
         # install and run 32 bit compiled extensions in that case.
-        result = "linux_i686"
+        machine = platform.machine()
+
+        if machine == "x86_64":
+            result = "linux_x32"
+        else:
+            result = "linux_i686"  # and machine == "i686"
```

Comment:
Maybe add a conditional with warning here, if `machine != "i686"`?

Reply:
Is there a standard way within pip that I should warn?

Reply:
`logger.warning`

Reply:
I don't think a warning is necessary here - falling back to source installs when there's no defined wheel tag is perfectly normal (e.g. it always happens on non-macOS BSD systems, Solaris, etc), and there's nothing stopping anyone setting up their own registry with `linux_x32` wheels if they want to (even if `PyPI` doesn't allow them).

Reply:
Sounds fair to me. :)

Reply:
The warning would be to highlight that the platform cannot be detected correctly; eg. some ficticious future x42 ABI that pip doesn't yet understand.

Reply:
Another possibility: if we set `result = "linux_" + machine`, would that be compatible with all platforms when using setarch? I guess I'd be worried about whether that formatting is right for all arches. (Since this patch will now make pip require setarch for non-x32 ABIs per https://github.com/pypa/pip/pull/5391#issuecomment-388672959)